design/one-pager-stacks-versioning.md

```diff
@@ -0,0 +1,248 @@
+# Stack Versioning
+* Owner: Luke Weber (@lukeweber)
+* Reviewers: Crossplane Maintainers
+* Status: Draft, revision 1.0
+
+## Problem
+
+Stacks installer currently relies on installing a particular version of stack from a docker tag. The stack installer in
+ the future will be able to support stack update policy, maintenance windows, warnings on major updates, and more.
+ To build more complex features around stack version management we will need a common way to express versions that is
+ widely understood and standards based, so we will unify on best practices and
+ [semantic version 2.0](https://semver.org/spec/v2.0.0.html) as the Crossplane versioning standard.
+
+#### Didn't docker tags already solve this?
+
+ * I don't want my versions underlying artifacts to change:
+    * Tags are a symbolic reference to an actual sha of an image and they can be updated or deleted
+ * I want to have context from a version:
+    * A sha is hard to compare when debugging
+      * Which is newer? - cb44dd0 or d2fb617?
+      * Is this alpha, beta?
+    * A tag like latest is opaque to the end user and so not incredibly useful:
+      * Which version am I running? you need to run special commands to figure out which sha is loaded
+      * on a node and then lookup that sha to see which version it's tagged against, if it's even still tagged.
+      * If the latest tag on this node the same as the latest tag on that node? When did it get restarted?
+
+As a note, if a user wants to continue to install from an arbitrary tag, stack-installer will work as expected, but
+ behavior relying on re-tagging latest and the stack installer detecting the tags underlying container being updated
+ will not trigger a re-install as the Crossplane stack installer by convention will assume that docker tags are
+ immutable and represent a single version. Any system that relies on tagging all builds with a :latest tag for example
+ is not recommended and in practice won't work as expected.
```

Comment:
> is not recommended and in practice won't work

Is this recommendation coming from a document or article thing that I can read if I want to know more about the background?

Reply:
I think there is an expectation that Stacks would use a `pullPolicy` like other image references in Kubernetes.  We haven't supported this in the past and I like that we are declaring that we are not going to.

This makes sense for Stacks because Stacks do not get scaled, destroyed and recreated, as Pods do.  The Stack Manager expects to install a particular version precisely one time and if that version of the Stack image does not change, the SM will do nothing else.